### PR TITLE
Add Week1-lecture0-Intro-to-NLP-Tasks

### LINKED ISSUE #42: Introduction to NLP Tasks
1. NER, Noun Phrase Extraction with spaCy
2. Text Classification with Random Forests/xgboost and TF-IDF
3. Explicitly call out that we don't recommend "old" NLP pipelines of stemming, stop word removal and then bag of words
4. Introduce over-fitting, training on test and data leaks
5. Demonstrate k-Fold Stratified Cross Validation here
6. Translation, Summarization with Transformers <- API Introduction, high level concept of a model

Show sample code from spacy to  perform all these tasks.

@RheagalFire please break this into smaller task

Filename: Week01-Lecture0-Introduction-to-NLP-Tasks.ipynb

### PR DESCRIPTION
**Pull request by RheagalFire**
### Description 
Add Intro to NLP Tasks Notebook 

closes #42

### CODE REVIEW

**Comment by RheagalFire** [comment]
<p>Understood. </p><p>Planning to give a outline(the format you suggested) first </p><p>then code + then other references which can optionally help understand the underlying logic. </p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554511'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by RheagalFire** [comment]
<p>🙌</p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554522'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by RheagalFire** [comment]
<p>Leaving it for you to decide , but since we have mentioned limitations of classical nlp pipeline , we can add/link references on how the attention mechanisms help in solving them. </p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554550'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by NirantK** [comment]
<p>Let's remove the attention intro and retain the task description and diagram</p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554550'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by NirantK** [comment]
Requested one example addition and then we're good to merge!

**Comment by RheagalFire** [comment]
<p>Two ways that can help here -&gt; </p><ol><li>Using Contextual Embeddings as a Features to the classical systems </li><li>Demonstrating Few Shot Classification with LLM and making prediction on the test set.</li></ol><p>Since this is supposedly lecture0 better to keep in mind not to expose too much content at once as we move on to cover these topics in details. </p><p> </p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2135117639'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by RheagalFire** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554540'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by RheagalFire** [comment]
<p>Removed the attention details , have kept the link to attention paper.</p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2132554550'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by RheagalFire** [comment]
<p>I've added 1 for now. LMK if you have something else in mind for this. </p>

---

 <a href='https://app.reviewnb.com/FuturePathAI/Learn-AI-Engineering/pull/49/discussion/#comment-2135117639'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>
